tests/ImageSharp.Tests/Formats/Png/PngDecoderTests.cs

```diff
@@ -142,20 +168,66 @@ public void Decode_Interlaced_ImageIsCorrect<TPixel>(TestImageProvider<TPixel> p
             }
         }
 
-        // TODO: We need to decode these into Rgba64 properly, and do 'CompareToOriginal' in a Rgba64 mode! (See #285)
         [Theory]
-        [WithFileCollection(nameof(TestImages48Bpp), PixelTypes.Rgba32)]
+        [WithFileCollection(nameof(TestImages48Bpp), PixelTypes.Rgb48)]
         public void Decode_48Bpp<TPixel>(TestImageProvider<TPixel> provider)
             where TPixel : struct, IPixel<TPixel>
         {
             using (Image<TPixel> image = provider.GetImage(new PngDecoder()))
             {
-                image.DebugSave(provider);
+                var encoder = new PngEncoder { ColorType = PngColorType.Rgb, BitDepth = PngBitDepth.Bit16 };
 
-                // Workaround a bug in mono-s System.Drawing PNG decoder. It can't deal with 48Bpp png-s :(
-                if (!TestEnvironment.IsLinux && !TestEnvironment.IsMono)
+                if (!SkipVerification(provider))
                 {
-                    image.CompareToOriginal(provider, ImageComparer.Exact);
+                    image.VerifyEncoder(provider, "png", null, encoder, customComparer: ValidatorComparer);
```

Comment:
Two calls to `DebugSave()` and `CompareToOriginal()` would more straightforward and readable in a decoder test. ( (`VerifyEncoder()` does the same though.)